plugin.rb

```diff
@@ -400,6 +480,31 @@ def self.notify(id)
 
       responses
     end
+
+    def self.migrate_legacy_data
+      rows = PluginStoreRow.where(plugin_name: PLUGIN_NAME).where("key ~* :pat", :pat => '^category_.*')
+
+      rows.each do |row|
+        ::PluginStore.cast_value(row.type_name, row.value).each do | rule |
+          unless rule[:migrated].present? && rule[:migrated]
+            id = row.key.gsub('category_', '')
+            channel = rule[:channel]
+            data = ::PluginStore.get(PLUGIN_NAME, "category_#{id}")
+            update = data.index { |i| i['channel'] === channel }
+            DiscourseSlack::Slack.set_filter(channel, rule[:filter], id)
+            data[update]['migrated'] = true
+            data = data.uniq { |i| i['channel'] }
+            ::PluginStore.set(PLUGIN_NAME, "category_#{id}", data.uniq)
+          end
+        end
+      end
+    end
+
+  end
+
+  unless ::PluginStore.get(PLUGIN_NAME, "legacy_migrated")
```

Comment:
Our practice of migrating the data format in plugins is to enqueue it in a `Onceoff` Job like https://github.com/discourse/discourse/blob/master/app/jobs/onceoff/grant_emoji.rb